### PR TITLE
Project setup to complete functionality with deployment

### PR DESCRIPTION
**Pull request by BilalLiaquat7**
In this PR: I have managed to setup project to complete the project with deployment.
multiple files are created along with functionality.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,182 @@
...
+- GitHub: [@githubhandle](https://github.com/BilalLiaquat7/)
+- LinkedIn: [@Linkedin](https://www.linkedin.com/in/muhammad-bilal-liaquat-87863390/)
+
+<p align="right">(<a href="#live-demo">back to top</a>)</p>
+
+## 🎥 Project Presentation
+
+- Project Demo: [Project Demo](https://metrics-8e7g.onrender.com/)
+
+- Project Presentation: [ProjectPresentation](https://www.loom.com/share/bf2e2616af3c4547af0b12303e332e39?sid=53ef421e-a178-407e-beab-dcd0c3371d85)
```
Comment (topeogunleye) [inline README.md]: - [ ] Great job on making your project presentation under 5 minutes and talking about giving a clear and objective description of the project in professional English, however please kindly add:

- a highlight about a piece of the code that you find interesting or are proud of making.

Remember, you're making good progress!. Keep up the good work!

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,15 @@
-import logo from './logo.svg';
+import { Route, Routes } from 'react-router-dom';
+import WeatherMain from './components/weatherMain';
+import WeatherDetail from './components/weatherDetail';
 import './App.css';
 
 function App() {
```
Comment (topeogunleye) [inline src/App.js:6]: Hi @BilalLiaquat7,

I noticed that your project lacks tests, both unit tests for individual components and integration tests using React Testing Library. According to the project requirements, tests are a crucial part of the submission and should not be overlooked.

Testing is a key aspect of production-grade code and contributes to its maintainability and robustness. Specifically, the requirements indicate that you should:

- Have unit tests for several components of the app.
- Include some pure functions in your Redux code and write tests for them.
- Perform integration tests using React Testing Library.
- Be capable of mocking data fetching and Redux Store connection.

Without tests, it is difficult to assess the reliability of the application. Please add the relevant tests to your project as specified in the requirements, and resubmit your project for another review.

If you have any questions about writing tests or need clarification on the project requirements, feel free to reach out. 

Best regards,
Temitope

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @BilalLiaquat7, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## (Highlights) Good Points: 
#### Creative Commons Attribution 👏

You've done an excellent job in giving proper credit to Nelson Sakwa, the original designer. The "Acknowledgements" section in the README adheres to the Creative Commons License requirements. Well done!

#### Live Version 🌐

Your project is deployed successfully using Render and both the home and detail pages work as expected. Adding a link to the deployed project in the README is a good touch. 🙌

#### Page Structure and Navigation 🗂️

Your project structure is organized logically, making it easy to navigate through your code. Your use of react-router-dom to handle page routing without reloads fits perfectly within the project requirements for a Single Page Application (SPA).

#### Technical Requirements 🛠️

Your implementation of Redux for state management is spot-on. You've correctly stored API data in the Redux store and connected it to your components. The filter component works seamlessly, adding a nice touch to the user experience.

#### Design 🎨

You have successfully followed the design guidelines, including typography and layout. The consistency in color scheme adds to the aesthetics of your app. Great job!

#### Overall 🌟

You have successfully met most of the project requirements, from code structure to technical specifications and design guidelines. Keep up the excellent work!

## Suggested Changes
_Check the comments under the review._

## Optional Changes
__You can also consider:__
- For future pull requests, please notice that your branch name `setup` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-setup`. This makes it easier to track the changes and the purpose of each branch.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (Davidosky007) [reply README.md]: * As mentioned by the previous CR Kindly highlight a piece of the code that you find interesting or are proud of making.

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @BilalLiaquat7 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* All tests passed 👍 
* Well written PR 👍 
* Live demo link works well 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (BilalLiaquat7) [reply README.md]: Is this mandatory to record a new video?
Couldn't be passed without it? Because I already recorded a video.

**Review by ademibowale — APPROVED** [review]
Hi @BilalLiaquat7  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
- Use functional components✔️
- follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))✔️
- The project contains a `Home` page and a `Detail` page✔️
- The category item detail page has its unique route within the SPA.
- Descriptive commit messages ✔️ 
- App has unit tests for several components of the app
- Good use of Github flow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**